distributed/utils.py

```diff
@@ -1543,3 +1543,27 @@ def clean_dashboard_address(addr, default_listen_ip=""):
         port = addr
 
     return {"address": host, "port": port}
+
+
+@functools.lru_cache(maxsize=256)
+def _clean_filename(filename):
+    """Cleans filename from sensitive information"""
+
+    if filename.startswith(sys.exec_prefix):
+        filename = filename.replace(sys.exec_prefix, "...")
+    elif filename.startswith(os.path.expanduser("~")):
+        filename = filename.replace(os.path.expanduser("~"), "")
+        filename = ".../" + "/".join(filename.split("/")[-3:])
```

Comment:
I think using only the last three parts of the path is too limiting. Some modules, for example `dask.dataframe`, will have a deeply nested hierarchy where only using the last three parts will remove valuable information:

```python
In [1]: import dask.dataframe as dd

In [2]: dd.read_parquet.__code__.co_filename
Out[2]: '/Users/james/projects/dask/dask/dask/dataframe/io/parquet/core.py'

In [3]: "/".join(dd.read_parquet.__code__.co_filename.split("/")[-3:])
Out[3]: 'io/parquet/core.py'
```

Reply:
Thanks James, that makes sense. Now do we want to put a cap on the amount of characters (before you suggested 80) which in this case will keep `/projects/dask/dask/dask/dataframe/io/parquet/core.py` (len =53) after removing the `os.path.expanduser("~")`. I'm thinking that 80 characters could end up bringing many directories that are not necessary useful and might be too long, but that depends on the length of the directories name.  Not sure how do we want to go about it...

Reply:
I'll roll back my previous suggestion about limiting the path in these cases. I suspect in the common case we'll be dealing with either Python stdlib modules or third-party packages in `site-packages`. We can always truncate things in the future if needed, but let's keep things simple for now.